dojo/importers/base_importer.py

```diff
@@ -0,0 +1,926 @@
+        if now is None:
+            now = timezone.now()
+        elif not now.tzinfo:
+            now = timezone.make_aware(now)
+
+        return scan_date, now
+
+    def get_user_if_supplied(
+        self,
+        user: Dojo_User = None,
+    ) -> Dojo_User:
+        """
+        Determines whether the user supplied at import time should
+        be used or not. If the user supplied is not actually a user,
+        the current authorized user will be fetched instead
+        """
+        if user is None:
+            return get_current_user()
+        return user
+
+    def verify_tool_configuration_from_test(
+        self,
+        api_scan_configuration: Tool_Configuration,
+        test: Test,
+    ) -> Test:
+        """
+        Verify that the Tool_Configuration supplied along with the
+        test is found on the product. If not, then raise a validation
+        error that will bubble up back to the user
+
+        if f there is a case where the Tool_Configuration supplied to
+        this function does not match the one saved on the test, then
+        we will user the one supplied rather than the one on the test.
+        """
+        # Do not bother with any of the verification if a Tool_Configuration is not supplied
+        if api_scan_configuration is None:
+            # Return early as there is no value in validating further
+            return test
+        # Ensure that a test was supplied
+        elif not isinstance(test, Test):
+            msg = "A test must be supplied to verify the Tool_Configuration against"
+            raise ValidationError(msg)
+        # Validate that the test has a value
+        elif test is not None:
+            # Make sure the Tool_Configuration is connected to the product that the test is
+            if api_scan_configuration.product != test.engagement.product:
+                msg = "API Scan Configuration has to be from same product as the Test"
+                raise ValidationError(msg)
+            # If the Tool_Configuration on the test is not the same as the one supplied, then lets
+            # use the one that is supplied
+            if test.api_scan_configuration != api_scan_configuration:
+                test.api_scan_configuration = api_scan_configuration
+                test.save()
+            # Return the test here for an early exit
+            return test
+
+    def verify_tool_configuration_from_engagement(
+        self,
+        api_scan_configuration: Tool_Configuration,
+        engagement: Engagement,
+    ) -> Test | Engagement:
+        """
+        Verify that the Tool_Configuration supplied along with the
+        engagement is found on the product. If not, then raise a validation
+        error that will bubble up back to the user
+
+        if f there is a case where the Tool_Configuration supplied to
+        this function does not match the one saved on the engagement, then
+        we will user the one supplied rather than the one on the engagement.
```

Comment:
This doesn't appear to be updated from the passed in `api_scan_configuration` the way it is on lines 704-706 above.

Reply:
This is how the original code was doing It, and I am nervous to change the behavior as part of this PR